src/ORM/Association/HasMany.php

```diff
@@ -198,7 +198,7 @@ public function saveAssociated(EntityInterface $entity, array $options = [])
      * target entity, and the parent entity.
      * @param \Cake\Datasource\EntityInterface $parentEntity The source entity containing the target
      * entities to be saved.
-     * @param \Cake\Datasource\EntityInterface[]|\Cake\Datasource\InvalidPropertyInterface[] $entities list of entities
+     * @param array $entities list of entities
```

Comment:
This needs to remain as `array`. The `foreach` loop for entities checks whether the array item is an `EntityInterface` instance, which means it could be something else too. We even have test for it https://github.com/cakephp/cakephp/blob/0cbace0fa29fda713f611f29135bdf7db443366d/tests/TestCase/ORM/Association/HasManyTest.php#L583.